includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1693,18 +1693,16 @@ private function is_amp_allowed_attribute( DOMAttr $attr_node, $attr_spec_list )
 		 * https://github.com/ampproject/amphtml/blob/1526498116488/extensions/amp-selector/validator-amp-selector.protoascii#L81-L91
 		 */
 		$descendant_reference_points = [
-			'amp-selector'         => AMP_Allowed_Tags_Generated::get_reference_point_spec( 'AMP-SELECTOR option' ),
-			'amp-story-grid-layer' => AMP_Allowed_Tags_Generated::get_reference_point_spec( 'AMP-STORY-GRID-LAYER default' ), // @todo Consider the more restrictive 'AMP-STORY-GRID-LAYER animate-in'.
+			'amp-selector'         => 'AMP-SELECTOR option',
+			'amp-story-grid-layer' => 'AMP-STORY-GRID-LAYER default', // @todo Consider the more restrictive 'AMP-STORY-GRID-LAYER animate-in'.
 		];
-		foreach ( $descendant_reference_points as $ancestor_name => $reference_point_spec ) {
+		foreach ( $descendant_reference_points as $ancestor_name => $reference_point_spec_name ) {
+			if ( empty( $this->open_elements[ $ancestor_name ] ) ) {
```

Comment:
This seems to do what the previous code did, so it's the same thing in faster.

However, I'd note that it is not technically correct, as the relationship between "some ancestor" and the descendant is not clear and could be invalid, depending on which nodes are in-between.

Reply:
I'm trying to think of an example that would be invalid. Do you have one in mind?